### PR TITLE
IGAPP-447: Additional field for feedback contact mail

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/routes/feedback/components/FeedbackModal.js [hunk 1]
```diff
@@ -60,33 +84,34 @@ class FeedbackModal extends React.Component<PropsType> {
...
-          <Input
-            theme={theme}
-            onChangeText={this.props.onCommentChanged}
-            autoFocus
-            value={comment}
-            multiline
-            placeholderTextColor={theme.colors.textSecondaryColor}
-            placeholder={t('yourFeedback')}
-          />
+          <Input theme={theme} onChangeText={props.onCommentChanged} value={comment} multiline />
```
Comment (klinzo) [inline native/src/routes/feedback/components/FeedbackModal.js:99]: Why did you remove the `autofocus` and `placeholder` props?

File: native/src/routes/feedback/components/FeedbackModal.js [hunk 2]
```diff
@@ -14,43 +14,67 @@ import buildConfig from '../../../modules/app/constants/buildConfig'
...
 `
 
 const RequiredText = styled.Text`
   color: red;
-  fontsize: 25px;
+  font-size: 25px;
+`
+
+const OptionalText = styled.Text`
+  font-weight: normal;
```
Comment (klinzo) [inline native/src/routes/feedback/components/FeedbackModal.js:48]: Is this necessary? Shouldn't that be the default?

File: native/src/routes/feedback/components/FeedbackModal.js [hunk 3]
```diff
@@ -14,43 +14,67 @@ import buildConfig from '../../../modules/app/constants/buildConfig'
...
+
+const Description = styled.Text`
+  display: flex;
+  text-align: left;
   color: ${props => props.theme.colors.textColor};
   font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  font-weight: bold;
 `
 
 const RequiredText = styled.Text`
```
Comment (klinzo) [inline native/src/routes/feedback/components/FeedbackModal.js:42]: I guess this is not used anymore, is it?

**Review by klinzo — COMMENTED** [review]
Some things that caught my eye:
* I think the red star is not necessary anymore as optional feedback is marked with `(optional)`
* I think there could be a little padding between the feedback mail and the send button
* I think the feedback mail input could be smaller in its height, currently it looks a little bit like a multi line input
* Do you think you could add tests for some of the behavior even though there are probably none right now?
* I think the feedback sent thanks message looked better without bold text.

**Comment by klinzo** [comment]
Tested on Android :+1:

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 1 above)
Comment (SteffiStuffel) [reply native/src/routes/feedback/components/FeedbackModal.js:99]: I oriented myself on the design in zeplin:
https://zpl.io/VQZPQek

Autofocus and placeholders are removed.

File: native/src/routes/feedback/components/FeedbackModal.js [hunk 4]
```diff
@@ -14,43 +14,67 @@ import buildConfig from '../../../modules/app/constants/buildConfig'
...
+  justify-content: space-between;
   padding: 15px 0 5px;
+`
+
+const Description = styled.Text`
+  display: flex;
+  text-align: left;
   color: ${props => props.theme.colors.textColor};
   font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  font-weight: bold;
```
Comment (Taggotty) [inline native/src/routes/feedback/components/FeedbackModal.js:39]: why did you explicitly set bold here?

File: native/src/routes/feedback/components/FeedbackModal.js [hunk 5]
```diff
@@ -60,33 +84,34 @@ class FeedbackModal extends React.Component<PropsType> {
...
-            placeholder={t('yourFeedback')}
-          />
+          <Input theme={theme} onChangeText={props.onCommentChanged} value={comment} multiline />
+
+          <DescriptionContainer theme={theme}>
+            <Description theme={theme}>{t('contactMailAddress')}</Description>
+            <OptionalText>({t('optionalInfo')})</OptionalText>
+          </DescriptionContainer>
+
+          <Input theme={theme} onChangeText={props.onFeedbackContactMailChanged} value={contactMail} />
```
Comment (Taggotty) [inline native/src/routes/feedback/components/FeedbackModal.js:106]: can you add some distance between this text and the button? I think it looks really weird on android

**Review by Taggotty — APPROVED** [review]

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 2 above)
Comment (SteffiStuffel) [reply native/src/routes/feedback/components/FeedbackModal.js:48]: yeah you are right. removed it.

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 3 above)
Comment (SteffiStuffel) [reply native/src/routes/feedback/components/FeedbackModal.js:42]: it was, but i removed it now.

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 4 above)
Comment (SteffiStuffel) [reply native/src/routes/feedback/components/FeedbackModal.js:39]: its part of the new design on zeplin

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 5 above)
Comment (SteffiStuffel) [reply native/src/routes/feedback/components/FeedbackModal.js:106]: done

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 1 above)
Comment (klinzo) [reply native/src/routes/feedback/components/FeedbackModal.js:99]: I'd keep the autofocus at least. I also liked the placeholder text.

**Review by maxammann — APPROVED** [review]

**Comment by SteffiStuffel** [comment]
> * I think the red star is not necessary anymore as optional feedback is marked with `(optional)` :heavy_check_mark: 
> * I think there could be a little padding between the feedback mail and the send button :heavy_check_mark: 
> * I think the feedback mail input could be smaller in its height, currently it looks a little bit like a multi line input :heavy_check_mark: 
> * Do you think you could add tests for some of the behavior even though there are probably none right now? :heavy_check_mark: 
> * I think the feedback sent thanks message looked better without bold text. :heavy_check_mark:

File: native/src/routes/feedback/components/FeedbackModal.js (hunk 1 above)
Comment (SteffiStuffel) [reply native/src/routes/feedback/components/FeedbackModal.js]: If we leave autofocus, then the keyboard is open when opening this site. When we remove it, I think it is a lot clearer, because when on opening the page you get an overview, which fields are there and see the send button. 
feedback
